### DISCUSSION TITLE
Warm up the meta cache on startup before accept connections

### DISCUSSION THREAD

**Discussion by sundy-li** [databendlabs/databend#4282] [Ideas]
I do think it's better to warm up the meta cache before serving any query.

As with any other database(MySQL, ClickHouse, Doris ...), metadatas are mostly stored locally, it will load them on startup then it's ready to accept connections.

**Comment by sundy-li** [comment]
Maybe we will have the cache stored inside the cloud services to speed up the start.

**Comment by bohutang** [comment]
Cache is hard in distributed system.
We already have `External Cache` task(PLANNING) on the [Roadmap 2022](https://github.com/datafuselabs/databend/issues/3706).
Before the `External Cache`, I would like suggest we can do some warm up policy by config:
1. Load table meta
2. Load table index
